### DISCUSSION TITLE
xcsoar_config.sh

### DISCUSSION THREAD

**Discussion by MaxKellermann** [Openvario/meta-openvario#234] [Ideas]
Why does https://github.com/Openvario/meta-openvario/blob/master/meta-ov/recipes-apps/ovmenu-ng-skripts/files/xcsoar_config.sh exist? What is its purpose?
This looks like complicated and fragile code, obtaining display rotation parameters from u-boot configuration, and patching all XCSoar profiles with it. But .... why??
By default, XCSoar keeps the current display rotation. Since ovmenu already operates at the correct rotation, what's the point of configuring the rotation again?

Related: why are there machine-specific files like this:
https://github.com/Openvario/meta-openvario/blob/e662707c8d777d113f738cca5088dcf84f1be67b/meta-ov/recipes-apps/ovmenu-ng/files/openvario-7-PQ070/machine.conf#L1
... to pass the display resolution as command-line parameter to XCSoar - when XCSoar by default auto-detects it? What's the point of that?

I'd like to delete all of this, but as always, I need to find out first whether this would break anything. There must be a reason why somebody went through all the trouble to write this code.

**Comment by mihu-ov** [comment]
IIRC these settings were required and there was no (working) autodetect with the first OV images so these command line parameters were included. Without specifying resolution XCSoar used  640x480 resolution. Don´t know if this was an error in the image, in XCSoar or in the configuration parameters.

**Reply by MaxKellermann** [reply]
XCSoar always had this auto-detection, so this should never have been necessary.
Maybe the proprietary Mali driver had bugs with that?
I guess I can remove this legacy cruft.
In any case, if anybody ever sees problems like that, it should be solved in XCSoar instead of having those kludges in OpenVario.

**Reply by MaxKellermann** [reply]
Really getting rid of kludges like that is problematic, because this profile-patching code leaves "broken" profiles behind which have explicit non-default rotation settings, and there's no way we can check whether these were user choice or just inserted by that script. What I'll do is pretend this (theoretical) problem doesn't exist and hope nobody gets annoyed.

**Reply by MaxKellermann** [reply]
Removed here: https://github.com/Openvario/meta-openvario/pull/237

**Comment by MaxKellermann** [comment]
btw. the rotation patching doesn't even work:
```
root@openvario-7-PQ070:~# cat /boot/config.uEnv 
rotation=0
root@openvario-7-PQ070:~# sh -x xcsoar_config.sh 
+ XCSOAR_CONFDIR=/home/root/.xcsoar/
+ echo 'XCSoar config file adaption'
XCSoar config file adaption
+ echo 'Mount config dir ...'
Mount config dir ...
+ mount /dev/mmcblk0p1 /boot
mount: /boot: /dev/mmcblk0p1 already mounted on /boot.
+ echo 'Doing DisplayOrientation ...'
Doing DisplayOrientation ...
+ display_rotation
++ grep rotation /boot/config.uEnv
+ TEMP=$'rotation=0\r'
+ '[' -n $'rotation=0\r' ']'
+ echo 'Config string found'
Config string found
+ ROTATION=$'\r'
+ echo Rotation $'\r'
Rotation 
+ case "$ROTATION" in
+ echo 'Replacing ... XCSOAR_ROTATION='
Replacing ... XCSOAR_ROTATION=
+ sed -i 's/^DisplayOrientation=.*/DisplayOrientation=""/' /home/root/.xcsoar//openvario.prf
+ echo 'Unmount config dir ...'
Unmount config dir ...
+ umount /dev/mmcblk0p1
+ echo 'Exiting ...'
Exiting ...
```
The script is bugged...
https://github.com/Openvario/meta-openvario/blob/42b5de934d3d95f76ca514c069cac90d01f1145b/meta-ov/recipes-apps/ovmenu-ng-skripts/files/xcsoar_config.sh#L13
This uses only the last byte from the line; but the last byte is a CR, because the file is in DOS text mode. It has been that way since the very first commit. Thus, I believe this feature has been broken forever, and it can be removed safely.

**Reply by kedder** [reply]
I believe this feature used to work, at least at some point.  It might be useless by now, because I think XCSoar got rotation autodetection (at least I remember implementing that).

**Reply by mihu-ov** [reply]
No idea how or why but rotation worked (display and touch) in the "old" images with 3.4 kernel. I am not sure if touchscreen also works on the newer 5.x kernel images

**Reply by MaxKellermann** [reply]
Maybe it worked before this file was converted to DOS text mode. Something broke when the "warrior" branch was created in 2019; whoever did it (commit author `build@build.fritz.box`) created lots of text files in DOS mode. The older branches do not have that problem.
One more reason why this kind of migration with a new branch was a mistake.
Anyway, I removed this `xcsoar_config.sh`, and display rotation just works, without any profile patching.

**Reply by mihu-ov** [reply]
Is the touch input rotated as well with the display rotation?
